packages/atoms/cal-provider/index.tsx

```diff
@@ -11,5 +11,30 @@ const ApiKeyContext = createContext("");
 export const useApiKey = () => useContext(ApiKeyContext);
 
 export function CalProvider({ apiKey, children }: CalProviderProps) {
-  return <ApiKeyContext.Provider value={apiKey}>{children}</ApiKeyContext.Provider>;
+  const [key, setKey] = useState("");
+
+  useEffect(() => {
+    async function verifyKey(key: string) {
+      try {
+        // here we'll call the /me endpoint in v2 to get user profile
+        // v2 is not ready yet thats why calling localhost for now
+        const response = await fetch(`http://localhost:3002/v1/me?apiKey=${key}`);
```

Comment:
you shouldn't hard code localhost:3002

Reply:
The original endpoint to be called would be something like `https://api.cal.com/v2/me` but since the v2 api is not ready, for the time being I was calling localhost